web/src/main/java/org/mskcc/cbio/oncokb/controller/DriveAnnotationParser.java

```diff
@@ -607,24 +609,45 @@ private static void parseTherapeuticImplications(Gene gene, Set<Alteration> alte
                 addDateToSetFromObject(lastEditDates, drugObj, "indication_review");
             }
 
-            String[] drugTxts = drugNameStr.replaceAll("(\\([^\\)]*\\))|(\\[[^\\]]*\\])", "").split(",");
-
             List<Treatment> treatments = new ArrayList<>();
-            for (int j = 0; j < drugTxts.length; j++) {
-                String[] drugNames = drugTxts[j].split(" ?\\+ ?");
+            for (int j = 0; j < therapiesArray.length(); j++) {
```

Comment:
j < therapiesArray.length();  Brackets here might be unnecessary.

Reply:
This is JAVA, and that's a method not property.

Reply:
I found the following words on https://www.geeksforgeeks.org/length-vs-length-java/
length vs length() in Java
array.length : length is a final variable applicable for arrays. With the help of length variable, we can obtain the size of the array.
string.length() : length() method is a final variable which is applicable for string objects. length() method returns the number of characters presents in the string.

I am not sure. Just a reminder. Sorry for it.

Reply:
@Yizhuzhu it depends on how the class is designed. It could be a public accessible property or a method. In these few places, JSONArray only implements the method instead of expose the property. I would not `worry` that much since the IDE will usually give you suggestions and it would not be compiled if that's actually the problem.